lib/SILOptimizer/Analysis/EscapeAnalysis.cpp

```diff
@@ -101,6 +101,49 @@ EscapeAnalysis::findRecursivePointerKind(SILType Ty,
   return NoPointer;
 }
 
+// Return the PointerKind that summarized an object's stored properties.
+//
+// If a class only holds fields of non-pointer types, then it is guaranteed not
+// to point to any other objects.
+EscapeAnalysis::PointerKind
+EscapeAnalysis::findObjectKind(SILType Ty, const SILFunction &F) const {
+  if (Ty.isAddress())
```

Comment:
Can this ever be an address? Operands to release instructions are not addresses and types of stored properties are not addresses either.

Reply:
I'm ambivalent about whether this should be an assert. It's ok right now to call it on any type and it gives you a conservative answer for non-class types.

Reply:
fine with me